pkg/kubernetes/ownerreference/v1alpha1/ownerreference.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	stringer "github.com/openebs/maya/pkg/apis/stringer/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OwnerReference is a wrapper over metav1.OwnerReference
+type OwnerReference struct {
+	Object *metav1.OwnerReference
+	errors []error
+}
+
+// String implements Stringer interface
+func (or OwnerReference) String() string {
+	return stringer.Yaml("ownerreference", or)
+}
+
+// GoString implements GoStringer interface
+func (or OwnerReference) GoString() string {
+	return or.String()
+}
+
+// New returns a new instance of OwnerReference
+func New() *OwnerReference {
+	return &OwnerReference{
+		Object: &metav1.OwnerReference{},
+	}
+}
+
+// WithName sets name in OwnerReference instance
+func (or *OwnerReference) WithName(name string) *OwnerReference {
```

Comment:
Let us make use of `Builder` as the struct that handles all these.
This helps in having a uniform standard that can be explained to new contributors.
In addition, this makes the code explicit to its use of common patterns.

Reply:
done PTAL.